terraform/stack_prod/provider.tf

```diff
@@ -4,6 +4,6 @@ provider "aws" {
   }
 
   region  = var.aws_region
-  version = "2.34.0"
+  version = "2.51.0"
```

Comment:
You can just use `~> 2.0`. I've never found locking TF stuff necessary as they version really well.